src/components/accessories/admission/admissionTable/AdmissionTable.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { FunctionComponent, useEffect } from "react";
+import Table from "../../table/Table";
+import { useTranslation } from "react-i18next";
+import InfoBox from "../../infoBox/InfoBox";
+import { CircularProgress } from "@material-ui/core";
+import { useDispatch, useSelector } from "react-redux";
+import { IState } from "../../../../types";
+import { AdmissionDTO } from "../../../../generated";
+import { getAdmissionsByPatientId } from "../../../../state/admissions/actions";
+import { format } from "date-fns";
+
+interface IOwnProps {
+  shouldUpdateTable: boolean;
+}
+
+const PatientAdmissionTable: FunctionComponent<IOwnProps> = ({
+  shouldUpdateTable,
+}) => {
+  const { t } = useTranslation();
+
+  const header = ["admDate"];
+
+  const label = {
+    admDate: t("admission.admDate"),
+    admType: t("admission.admType"),
+    diseaseIn: t("admission.diseaseIn"),
+    transUnit: t("admission.transUnit"),
+    ward: t("admission.ward"),
+  };
+  const order = ["admDate"];
+
+  const dispatch = useDispatch();
+
+  const data = useSelector<IState, AdmissionDTO[]>((state) =>
+    state.admissions.admissionsByPatientId.data
+      ? state.admissions.admissionsByPatientId.data
+      : []
+  );
+
+  const patientCode = useSelector<IState, number | undefined>(
+    (state) => state.patients.selectedPatient.data?.code
+  );
+
+  useEffect(() => {
+    if (shouldUpdateTable || patientCode) {
+      dispatch(getAdmissionsByPatientId(patientCode));
+    }
+  }, [shouldUpdateTable, dispatch, patientCode, getAdmissionsByPatientId]);
+
+  const formatDataToDisplay = (data: AdmissionDTO[]) => {
+    return data.map((item) => {
+      return {
+        admDate: item.admDate
+          ? format(
+              new Date(isNaN(+item.admDate) ? item.admDate : +item.admDate),
```

Comment:
I can't understand this. If adminDate is NaN you pass adminDate in any case, right?

Reply:
I replaced it by new Date(+item.admDate)

Reply:
as we are expecting a milliseconds date.